reference/test/base/batch_multi_vector_kernels.cpp

```diff
@@ -0,0 +1,425 @@
+        {{{2.0, -2.0, 1.5}}, {{2.0, -2.0, 3.0}}}, this->exec);
+    auto ualpha = alpha->unbatch();
+
+    this->mtx_1->add_scaled(alpha.get(), this->mtx_0.get());
+    this->mtx_10->add_scaled(ualpha[0].get(), this->mtx_00.get());
+    this->mtx_11->add_scaled(ualpha[1].get(), this->mtx_01.get());
+
+    auto res = this->mtx_1->unbatch();
+    GKO_ASSERT_MTX_NEAR(res[0].get(), this->mtx_10.get(), 0.);
+    GKO_ASSERT_MTX_NEAR(res[1].get(), this->mtx_11.get(), 0.);
+}
+
+
+TYPED_TEST(BatchMultiVector, AddsScaledWithScalar)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using T = typename TestFixture::value_type;
+    auto alpha = gko::batch_initialize<Mtx>({{2.0}, {-2.0}}, this->exec);
+    auto ualpha = alpha->unbatch();
+
+    this->mtx_1->add_scaled(alpha.get(), this->mtx_0.get());
+    this->mtx_10->add_scaled(ualpha[0].get(), this->mtx_00.get());
+    this->mtx_11->add_scaled(ualpha[1].get(), this->mtx_01.get());
+
+    auto res = this->mtx_1->unbatch();
+    GKO_ASSERT_MTX_NEAR(res[0].get(), this->mtx_10.get(), 0.);
+    GKO_ASSERT_MTX_NEAR(res[1].get(), this->mtx_11.get(), 0.);
+}
+
+
+TYPED_TEST(BatchMultiVector, AddScaledFailsOnWrongSizes)
+{
+    using Mtx = typename TestFixture::Mtx;
+    auto alpha = gko::batch_initialize<Mtx>(
+        {{2.0, 3.0, 4.0, 5.0}, {-2.0, 2.0, 4.0, 5.0}}, this->exec);
+
+    ASSERT_THROW(this->mtx_1->add_scaled(alpha.get(), this->mtx_2.get()),
+                 gko::DimensionMismatch);
+}
+
+
+TYPED_TEST(BatchMultiVector, ComputesDot)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using T = typename TestFixture::value_type;
+    auto result =
+        Mtx::create(this->exec, gko::batch_dim<2>(2, gko::dim<2>{1, 3}));
+    auto ures = result->unbatch();
+
+    this->mtx_0->compute_dot(this->mtx_1.get(), result.get());
+    this->mtx_00->compute_dot(this->mtx_10.get(), ures[0].get());
+    this->mtx_01->compute_dot(this->mtx_11.get(), ures[1].get());
+
+    auto res = result->unbatch();
+    GKO_ASSERT_MTX_NEAR(res[0].get(), ures[0].get(), 0.);
+    GKO_ASSERT_MTX_NEAR(res[1].get(), ures[1].get(), 0.);
+}
+
+
+TYPED_TEST(BatchMultiVector, ComputeDotFailsOnWrongInputSize)
+{
+    using Mtx = typename TestFixture::Mtx;
+    auto result =
+        Mtx::create(this->exec, gko::batch_dim<2>(2, gko::dim<2>{1, 3}));
+
+    ASSERT_THROW(this->mtx_1->compute_dot(this->mtx_2.get(), result.get()),
+                 gko::DimensionMismatch);
+}
+
+
+TYPED_TEST(BatchMultiVector, ComputeDotFailsOnWrongResultSize)
+{
+    using Mtx = typename TestFixture::Mtx;
+    auto result =
+        Mtx::create(this->exec, gko::batch_dim<2>(2, gko::dim<2>{1, 2}));
+    auto result2 =
+        Mtx::create(this->exec, gko::batch_dim<2>(2, gko::dim<2>{1, 2}));
+
+    ASSERT_THROW(this->mtx_0->compute_dot(this->mtx_1.get(), result.get()),
+                 gko::DimensionMismatch);
+    ASSERT_THROW(this->mtx_0->compute_dot(this->mtx_1.get(), result2.get()),
+                 gko::DimensionMismatch);
```

Comment:
Aren't `result` and `result2` the same?
```suggestion
    auto result =
        Mtx::create(this->exec, gko::batch_dim<2>(2, gko::dim<2>{1, 2}));

    ASSERT_THROW(this->mtx_0->compute_dot(this->mtx_1.get(), result.get()),
                 gko::DimensionMismatch);
```
same for conj_dot